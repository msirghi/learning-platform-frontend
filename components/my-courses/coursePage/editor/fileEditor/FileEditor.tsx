import React from 'react';
import styles from '../CourseEditor.module.scss';
import InsertDriveFileIcon from '@material-ui/icons/InsertDriveFile';
import ClearIcon from '@material-ui/icons/Clear';
import AddIcon from '@material-ui/icons/Add';
import { Button } from '@material-ui/core';
import { useTranslation } from '../../../../../i18n';

type Props = {
  selectedFiles: File[];
  onFileSelect: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onFileRemove: (e: File) => void;
};

export const FileEditor: React.FC<Props> = ({ selectedFiles, onFileSelect, onFileRemove }) => {
  const { t } = useTranslation();

  return (
    <div className={styles.attachementsContainer}>
      <div className={styles.sectionTitle}>{t('courses:attachFilesLabel')}</div>
      {selectedFiles.length !== 0 &&
        selectedFiles.map((file, idx) => {
          return (
            <div key={idx} className={styles.singleFileContainer}>
              <InsertDriveFileIcon color='primary' />
              <span>{file.name}</span>
              <ClearIcon
                onClick={() => onFileRemove(file)}
                className={styles.deleteIcon}
                color='secondary'
                fontSize='small'
              />
            </div>
          );
        })}
      <input
        color='primary'
        type='file'
        id='icon-button-file'
        onChange={(e) => onFileSelect(e)}
        style={{ display: 'none' }}
      />
      <label htmlFor='icon-button-file'>
        <Button
          className={styles.addFileButton}
          component='span'
          startIcon={<AddIcon />}
          size='large'
          color='primary'
        >
          {t('courses:addFileLabel')}
        </Button>
      </label>
    </div>
  );
};
