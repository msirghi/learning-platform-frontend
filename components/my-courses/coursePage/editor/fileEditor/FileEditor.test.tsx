import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { FileEditor } from './FileEditor';
import ClearIcon from '@material-ui/icons/Clear';

describe('FileEditor component', () => {
  global.File = class MockFile {
    filename: string;
    constructor(
      parts: (string | Blob | ArrayBuffer | ArrayBufferView)[],
      filename: string,
      properties?: FilePropertyBag
    ) {
      this.filename = filename;
    }
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const defaultProps = {
    selectedFiles: [new global.File(('' as unknown) as BlobPart[], 'New file.txt')],
    onFileSelect: jest.fn(),
    onFileRemove: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<FileEditor {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render invisible input', () => {
    const wrapper = shallow(<FileEditor {...defaultProps} />);
    const input = wrapper.find('input');
    expect(input).toHaveLength(1);
    expect(input.props().style.display).toEqual('none');
  });

  it('should fire prop function on adding the file', () => {
    const wrapper = shallow(<FileEditor {...defaultProps} />);
    const input = wrapper.find('input');
    expect(input).toHaveLength(1);

    input.simulate('change');
    expect(defaultProps.onFileSelect).toHaveBeenCalledTimes(1);
  });

  it('should fire prop function on removing the file', () => {
    const wrapper = shallow(<FileEditor {...defaultProps} />);
    const btn = wrapper.find(ClearIcon);
    expect(btn).toHaveLength(1);

    btn.simulate('click');
    expect(defaultProps.onFileRemove).toHaveBeenCalledTimes(1);
  });
});
