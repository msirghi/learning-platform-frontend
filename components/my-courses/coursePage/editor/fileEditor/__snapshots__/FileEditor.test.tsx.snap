// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileEditor component should match the snapshot 1`] = `
<div
  className="attachementsContainer"
>
  <div
    className="sectionTitle"
  >
    courses:attachFilesLabel
  </div>
  <div
    className="singleFileContainer"
    key="0"
  >
    <Memo([object Object])
      color="primary"
    />
    <span />
    <Memo([object Object])
      className="deleteIcon"
      color="secondary"
      fontSize="small"
      onClick={[Function]}
    />
  </div>
  <input
    color="primary"
    id="icon-button-file"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <label
    htmlFor="icon-button-file"
  >
    <WithStyles(ForwardRef(Button))
      className="addFileButton"
      color="primary"
      component="span"
      size="large"
      startIcon={<Memo />}
    >
      courses:addFileLabel
    </WithStyles(ForwardRef(Button))>
  </label>
</div>
`;
